tiddit: init at 3.6.1

apraga opened the pull request:
Genomic tool to call structural variations.

## Description of changes

New package : a CLI tool written in python. It depends on 2 other executables, not packaged as a separate derivation.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fabaff commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,75 @@
+  python3Packages,
+}:
+let
+  fermi2 = stdenv.mkDerivation {
```
Move `fermi2` to it's own derivation.

fabaff commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,75 @@
+let
+  fermi2 = stdenv.mkDerivation {
+    pname = "fermi2";
+    version = "unstable-20210521";
```
```suggestion
    version = "0-unstable-2021-05-21";
```

see https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

fabaff commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,75 @@
+      cp fermi2 $out/bin
+    '';
+  };
+  ropebwt2 = stdenv.mkDerivation {
```
Move `ropebwt2` to it's own derivation.

fabaff commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,75 @@
+  };
+  ropebwt2 = stdenv.mkDerivation {
+    name = "ropebwt2";
+    version = "unstable-20210201";
```
```suggestion
    version = "0-unstable-2021-02-01";
```

fabaff commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,75 @@
+in
+python3Packages.buildPythonApplication {
+  name = "tiddit";
+  version = "3.6.1";
+  src = fetchFromGitHub {
```
```suggestion
  version = "3.6.1";
  pyproject = true;

  src = fetchFromGitHub {
```

and add later `build-system = ...`

apraga replied:
What is the advantage of pyproject here, where there is only a setup.py ? The manual says it default to setuptools. Thanks,

fabaff commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,75 @@
+    homepage = "https://github.com/SciLifeLab/TIDDIT";
+    description = "Identify chromosomal rearrangements using Mate Pair or Paired End sequencing data";
+    mainProgram = "tiddit";
+    license = licenses.gpl3;
```
I can't find details if the license is `GPLv3+`, thus we should consider it `GPLv3`.

```suggestion
    license = licenses.gpl3Only;
```

apraga commented:
Thanks for your review (and fast too). I've updated the PR accordingly. Please tell me if something is missing.

apraga commented:
@fabaff This PR should be ready to merge and your comment should have been addressed. If you could look at it, it would be nice :)

apraga commented:
@fabaff Hi, I've rebased the change against master, removed useless arguments and applied nixfmt. Does that looks good to you ? Thanks

Aleksanaa commented:
Please check ofborg build failure below. If you don't have the device and there's no clue to fix them upstream (issues/prs), please mark these platforms as broken.

paparodeo commented on pkgs/by-name/fe/fermi2/package.nix:
```diff
@@ -0,0 +1,33 @@
+  };
+
+  buildInputs = [ zlib ];
+  installPhase = ''
```
use default compiler for platform. this fixes platforms that use clang and not gcc
```suggestion
  makeFlags = [ "CC=${stdenv.cc.targetPrefix}cc" ];
  installPhase = ''
```

paparodeo commented on pkgs/by-name/fe/fermi2/package.nix:
```diff
@@ -0,0 +1,33 @@
+    mainProgram = "fermi2";
+    license = licenses.mit;
+    maintainers = with maintainers; [ apraga ];
+    platforms = platforms.unix;
```
I think this is only supported by x86-64 so that needs to be specified.

paparodeo commented on pkgs/by-name/ro/ropebwt2/package.nix:
```diff
@@ -0,0 +1,33 @@
+  };
+
+  buildInputs = [ zlib ];
+  installPhase = ''
```
```suggestion
  makeFlags = [ "CC=${stdenv.cc.targetPrefix}cc" ];
  installPhase = ''
```

apraga replied:
Okay, I've added the changes, thanks a lot !

paparodeo reviewed (commented):
this is a clang vs gcc issue. can use clang on linux like:
```
nix-build -E 'with import ./.{}; fermi2.override { stdenv = clangStdenv; }'
```

paparodeo reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 312995`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>fermi2</li>
    <li>ropebwt2</li>
    <li>tiddit</li>
    <li>tiddit.dist</li>
  </ul>
</details>

---

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 312995`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>fermi2</li>
    <li>ropebwt2</li>
    <li>tiddit</li>
    <li>tiddit.dist</li>
  </ul>
</details>

fabaff reviewed (commented):
Could you please update the commit messages for `ropebwt2` and `fermi2`?

apraga commented:
I've updated the commits to reflect those changes. Does that looks good to you ?

paparodeo commented:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions
the first line  for new packages should be `fermi2: init 0.1-unstable-2021-05-21`

apraga commented:
Sorry about that, commits are corrected :/

Aleksanaa commented on pkgs/by-name/fe/fermi2/package.nix:
```diff
@@ -0,0 +1,37 @@
+
+  makeFlags = [ "CC=${stdenv.cc.targetPrefix}cc" ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp fermi2 $out/bin
+  '';
```
```suggestion
  installPhase = ''
    runHook preInstall
    
    install -Dm755 fermi2 -t $out/bin
    
    runHook postInstall
  '';
```

Aleksanaa commented on pkgs/by-name/fe/fermi2/package.nix:
```diff
@@ -0,0 +1,37 @@
+    cp fermi2 $out/bin
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/lh3/fermi2";
+    description = "Successor of fermi, a whole genome de novo assembler based on the FMD-index for large genomes ";
+    mainProgram = "fermi2";
+    license = licenses.mit;
+    maintainers = with maintainers; [ apraga ];
```
```suggestion
  meta = {
    homepage = "https://github.com/lh3/fermi2";
    description = "Successor of fermi, a whole genome de novo assembler based on the FMD-index for large genomes ";
    mainProgram = "fermi2";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ apraga ];
```

Aleksanaa commented on pkgs/by-name/fe/fermi2/package.nix:
```diff
@@ -0,0 +1,37 @@
+
+  meta = with lib; {
+    homepage = "https://github.com/lh3/fermi2";
+    description = "Successor of fermi, a whole genome de novo assembler based on the FMD-index for large genomes ";
```
```suggestion
    description = "Successor of fermi, a whole genome de novo assembler based on the FMD-index for large genomes";
```

Aleksanaa commented on pkgs/by-name/ro/ropebwt2/package.nix:
```diff
@@ -0,0 +1,34 @@
+
+  buildInputs = [ zlib ];
+  makeFlags = [ "CC=${stdenv.cc.targetPrefix}cc" ];
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ropebwt2 $out/bin
+  '';
```
```suggestion
  installPhase = ''
    runHook preInstall
    
    install -Dm755 ropebwt2 -t $out/bin
    
    runHook postInstall
  '';
```

Aleksanaa commented on pkgs/by-name/ro/ropebwt2/package.nix:
```diff
@@ -0,0 +1,34 @@
+    cp ropebwt2 $out/bin
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/lh3/ropebwt2";
+    description = "Incremental construction of FM-index for DNA sequences";
+    mainProgram = "ropebwt2";
+    license = licenses.mit;
+    maintainers = with maintainers; [ apraga ];
+    platforms = platforms.unix;
```
```suggestion
  meta = {
    homepage = "https://github.com/lh3/ropebwt2";
    description = "Incremental construction of FM-index for DNA sequences";
    mainProgram = "ropebwwith lib; t2";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ apraga ];
    platforms = lib.platforms.unix;
```

Aleksanaa commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,44 @@
+  python3Packages,
+  stdenv,
+}:
+python3Packages.buildPythonApplication {
+  name = "tiddit";
+  version = "3.6.1";
+  pyproject = true;
+  src = fetchFromGitHub {
+    owner = "SciLifeLab";
+    repo = "TIDDIT";
+    rev = "TIDDIT-3.6.1";
+    hash = "sha256-OeqVQJDw0fmSDWIGab2qtTJCzZxqLY2XzRqaTRuPIdI=";
+  };
```
```suggestion
python3Packages.buildPythonApplication rec {
  name = "tiddit";
  version = "3.6.1";
  pyproject = true;

  src = fetchFromGitHub {
    owner = "SciLifeLab";
    repo = "TIDDIT";
    rev = "refs/tags/TIDDIT-${version}";
    hash = "sha256-OeqVQJDw0fmSDWIGab2qtTJCzZxqLY2XzRqaTRuPIdI=";
  };
```

Aleksanaa commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,44 @@
+    joblib
+    numpy
+    pysam
+  ];
```
```suggestion
  ];

```

Aleksanaa commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,44 @@
+    numpy
+    pysam
+  ];
+  propagatedBuildInputs = [
+    fermi2
+    ropebwt2
+  ];
```
```suggestion
  makeWrapperArgs = [ "--prefix PATH : ${lib.makeBinPath [ fermi2 ropebwt2 ]}" ];
```

Test it yourself

Aleksanaa commented on pkgs/by-name/ti/tiddit/package.nix:
```diff
@@ -0,0 +1,44 @@
+    ropebwt2
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/SciLifeLab/TIDDIT";
+    description = "Identify chromosomal rearrangements using Mate Pair or Paired End sequencing data";
+    mainProgram = "tiddit";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ apraga ];
+    platforms = platforms.unix;
```
```suggestion
  meta = {
    homepage = "https://github.com/SciLifeLab/TIDDIT";
    description = "Identify chromosomal rearrangements using Mate Pair or Paired End sequencing data";
    mainProgram = "tiddit";
    license = lib.licenses.gpl3Only;
    maintainers = with lib.maintainers; [ apraga ];
    platforms = lib.platforms.unix;
```

apraga commented:
Thanks, your review has been integrand and rebased (hopefully without messing anything up). I've added `bwa` to the PATH as it's a requiried runtime dependency (somewhat got lost in the process).

apraga committed:
tiddit: init at 3.6.1

Genomic tool to call structural variations.

This packages requires 2 executables,packaging fermi2, bwt2, that are
packaged in the same derivation and added to the path later on.

apraga committed:
ropebwt2: init at 0-unstable-2021-02-01

Tiddit dependency. GCC use is hardcoded in Makefile so CC is overridden
to use clang on other platforms.

apraga committed:
fermi2: init at unstable-2021-05-21

Tiddit dependency.

apraga commented:
While rebasing, I removed the commit for ropebwt2 and fermi2. They have been added back by picking older commits and checking your reviews as well.

FliegendeWurst commented on pkgs/by-name/fe/fermi2/package.nix:
```diff
@@ -0,0 +1,33 @@
+  zlib,
+  stdenv,
+}:
+stdenv.mkDerivation {
```
Please override to use gccStdenv on Darwin. (Or patch it to use clang)
```
Running phase: buildPhase
cc -c -g -Wall -O2    rle.c -o rle.o
build flags: SHELL=/nix/store/fhngvmgagwwxhak8nzrx5vqbg4g4qiac-bash-5.2p37/bin/bash
gcc -c -g -Wall -O2 -Wno-unused-function    kthread.c -o kthread.o
/nix/store/fhngvmgagwwxhak8nzrx5vqbg4g4qiac-bash-5.2p37/bin/bash: line 1: gcc: command not found
```

apraga replied:
I have lost the edits that fixed that. `makeFlags` have been updated. Sorry :/

FliegendeWurst commented on pkgs/by-name/fe/fermi2/package.nix:
```diff
@@ -0,0 +1,33 @@
+
+    runHook postInstall
+  '';
+  meta = {
```
Please mark as broken for aarch64-linux or fix
```
gcc -c -g -Wall -O2 -Wno-unused-function    ksw.c -o ksw.o
ksw.c:28:10: fatal error: emmintrin.h: No such file or directory
   28 | #include <emmintrin.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:13: ksw.o] Error 1
```

apraga replied:
Same here: adding back `    platforms = lib.intersectLists lib.platforms.x86_64 lib.platforms.unix;`
